File: app/src/main/java/com/example/zhouzhihui/audioshift/dagger/MainComponent.java
package com.example.zhouzhihui.audioshift.dagger;

import com.example.zhouzhihui.audioshift.MainActivity;

import javax.inject.Singleton;

import dagger.Component;

@Component(modules = MainModule.class)
@Singleton
public interface MainComponent {
    void inject(MainActivity mainActivity);
}

File: app/src/main/java/com/example/zhouzhihui/audioshift/dagger/MainComponent.kt
package com.example.zhouzhihui.audioshift.dagger

import com.example.zhouzhihui.audioshift.MainActivity

import javax.inject.Singleton

import dagger.Component

@Component(modules = arrayOf(MainModule::class))
@Singleton
interface MainComponent {
    fun inject(mainActivity: MainActivity)
}
